src/agent/Cargo.toml

```diff
@@ -87,9 +95,12 @@ members = [
 lto = true
 
 [features]
+# The default-pull feature would support all pull types, including sharing images by virtio-fs and pulling images in the guest
```

Comment:
Hey @ChengyuZhu6 - I've probably missed this part of the conversation, but I'm interested in how `default-pull` and `guest-pull` are different. Is it just to set us up in future to have different options?

Reply:
> > I think the idea in this PR is probably to avoid cluttering the non-CoCo kata agent with the image pull code, but I guess you are suggesting that we use the same agent binary for CoCo and Kata?
> 
> Yep, otherwise we'd have to have:
> 
> * kata-agent
> * kata-agent-opa
> * kata-agent-opa-guest-pull
> * kata-agent-opa-host-pull
> 
> And for each kata-agent-opa-* we'd have to have:
> 
> * kata-containers-confidential-guest-pull.img
> * kata-containers-confidential-guest-pull-initrd.img
> * kata-containers-confidential-host-pull.img
> * kata-containers-confidential-host-pull-initrd.img
> 
> And then test all the combinations of those, and this simply doesn't escale.
> 
> So, my suggestion here is, let's find a way, even if it's passing pull-type=foo either via policy or in the kernel command line (via agent.pull-type, which has to be introduced), and then make this a run time condition rathr and build time one.
> 
> > It is feasible to determine the behavior at runtime. The old approach of checking if `AA_KBC_PARAMS` is set is a little weird (maybe it would have been clearer if we'd had a separate flag to enable image pull). There has been some discussion of using InitData to setup an image-rs configuration file. This brings us back to the policy vs initdata discussion, but either way there is actually a bit of a snag. This setting is something we would definitely have to support on platforms that don't support initdata measurement. Maybe a kernel command line flag would be better after all? I'd have to think about it more.
> 
> pull-type can be the name of the parameter passed, that would make things cleaner.

> As you can see, the agent that supports guest pull will download and compile about two hundred more dependencies, which results in the agent that supports guest pull being twice the size of another agent. I am not sure whether this will cause any issues when merging the code into main. If there are no issues, I will make the changes to support direct compilation of agents that support guest-pull and kata-host-share.

This is definitely not optimal, but still better than having to select the agent pull-type at build time.

If you want to offer the option for users to only choose one specific pull-type, I'd be okay with that, as long as there's a pull-type=all option, and that's what we'll be using and shipping by default.  Then if users want to optmise the size, they can do, but not at the cost of making our Ci more complex.

@stevenhorsman `default-pull` feature addresses the issue raised by Fabiano to make things cleaner on CI. Meanwhile, `guest-pull` option is available for those who prefer not to integrate image pull code into the non-CoCo kata agent, maintaining a clutter-free environment.